pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +55,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = WrapClientErrors(err)
```

Comment:
I get your explanation, but there is a reason for the complexity of this piece. I really wish it was as simple as just displaying a different message for end-users of `osc` when I wrote it.  :)
 
Take for example: user downloads `osc` and runs `osc get pods` right away. We don't want to just display a "no server found for" message, we want to say "Welcome, we are OpenShift, you are not configured, need to run run osc login, blah". 

But *we don't have access to the places where `kubectl`, which `get` belongs to, prints its messages*. So we can't just return the string, we must propagate the error with the pretty string replacing the original one until it gets printed. `kubectl` even does `Fatalf` in some places.

So it seems a bigger problem. One possible solution would be to every command (including upstream) never print errors right away, but instead always propagate them and let `Execute()` decide what to do about it (in our case, replace string with prettier ones, and only then `Fatalf`).